File: with-lombok/src/test/java/TestBuilder.java
import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import lombok.Builder;
import lombok.ToString;

class TestBuilder {

    @Builder
    @ToString
    static
    class MyBigClass{
        private int attr1;
        private int attr2;
        private int attr3;
        private int attr4;
        private int attr5;
        private int attr6;
        private int attr7;
        private int attr8;
        private int attr9;
    }

    @Test
    @DisplayName("should build a partial instance of MyBigClass")
    void shouldBuildAnInstance(){
        MyBigClass myInstance = MyBigClass.builder()
                .attr3(42)
                .attr5(99)
                .build();

        Assertions.assertThat(myInstance)
                .hasToString("TestBuilder.MyBigClass("
                        + "attr1=0, attr2=0, attr3=42, attr4=0, attr5=99, "
                        + "attr6=0, attr7=0, attr8=0, attr9=0)");
    }
}

File: with-kotlin/src/test/kotlin/TestBuilder.kt
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

/**
 * No native equivalent for @Builder in Kotlin.
 * Use constructor with named parameter instead.
 */
class TestBuilder {

    data class MyBigClass(
        var attr1: Int = 0,
        var attr2: Int = 0,
        var attr3: Int = 0,
        var attr4: Int = 0,
        var attr5: Int = 0,
        var attr6: Int = 0,
        var attr7: Int = 0,
        var attr8: Int = 0,
        var attr9: Int = 0,
    )

    @Test
    fun `should build a partial an instance`() {
        val instance = MyBigClass(attr3 = 42, attr5 = 99)
            .apply {
                attr8 = 888
            }

        assertEquals(
            "MyBigClass(attr1=0, attr2=0, attr3=42, attr4=0, " +
                    "attr5=99, attr6=0, attr7=0, attr8=888, attr9=0)",
            instance.toString()
        )
    }
}
